engine/src/main/java/org/hibernate/search/engine/backend/index/spi/IndexingMonitor.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.backend.index.spi;
+
+public interface IndexingMonitor {
```

Comment:
If possible, I'd prefer this interface to be in the ORM module, next to the mass indexer. Maybe also rename it `MassIndexingMonitor`?
As far as I can tell, it should be possible, since all these methods will be called directly from the mass indexer.

Also, this should be API, not SPI: users are expected to implement it for their own purposes (displaying a progress bar in their web application, for example).

Reply:
OK Even if I think that the monitor feature could be used for a general purpose tracing, very common these days. No problem to confine the monitor to the ORM module at the moment, we can refactor it later on.